Implement SVG renderer

matteodelabre opened the pull request:
This is a new renderer (`lines2svg`) able to create SVG files based on `.lines` files. One file is created per page, each named `test-n.png`, where `n` is the index of the page.

_Implemented features:_

* basic, non-textured strokes;
* base brush size;
* color;
* highlighter;
* normal eraser and region eraser.

_Unsupported features:_

* textures;
* base size variation based on pressure and tilt.

The features I didn’t implement are going to be a bit trickier to integrate into SVG in my opinion, because it doesn’t provide a way to vary the width of a stroke along a line. It might be possible to render textures using a SVG `<pattern>`.

I tested the new renderer with [this notebook](https://github.com/ax3l/lines-are-beautiful/files/2247383/svg-samples.zip): I don’t know if it’s of interest to be committed to the repo?

matteodelabre committed:
Add SVG renderer `lines2svg`

matteodelabre committed:
Fix build on older GCC versions

ax3l commented on include/rmlab/renderer/lines2svg.cpp:
```diff
@@ -0,0 +1,339 @@
+/* Copyright 2017-2018 Axel Huebl
```
@matteodelabre pls feel free to add your name to this file :)
(Is it ok if in ASCII? I have weird experiences with some compilers with non-ascii comments in source files)

matteodelabre replied:
Alright! I can second the many problems with non-ASCII chars in code. Maybe someday we’ll be able to use Unicode everywhere without thinking too much about it :-)

ax3l replied:
Haha, thanks. Same with my last name ;)

ax3l commented:
Feel free to add the notebook to `share/rmlab/examples/` :-)

ax3l reviewed (commented):
That's so cool! Added a small inline comment.

I am currently digging deeper in exact pixel-by-pixel rendering for various brushes. If you like we can chat a little in the evenings/weekends on details.

ax3l commented:
Uh, pls add the new tool in the `README.md` below the `lines2png` tool in the **CLI** section :)

matteodelabre committed:
Add testing notebook & pre-rendered images

matteodelabre committed:
Add myself to copyright notice in `lines2svg`

matteodelabre committed:
Update README with info about the SVG renderer

ax3l commented:
Haha, thanks. Same with my last name ;)

matteodelabre commented:
I added the test notebook and updated the README! It’d be interesting to chat about the rendering: for this tool, most values, especially line widths, result from guesswork. I’m in Montreal, Canada (UTC-4) until September, which will probably make it difficult to find a common time during the week, but for weekends it’ll probably be fine.

matteodelabre committed:
Fix size_t initializers

ax3l reviewed (approved):
Thank you, that's great!
Just a quick question: the `.rendered/` dir contains "original" renderings from the official software for comparisons, right?

matteodelabre commented:
Yup, those are the PNG files found in the `.cache` directories. They provide a useful reference while developing renderers.

ax3l commented:
Absolutely agree, just wanted to make sure they are the reference ones :)

matteodelabre commented:
Hi @ax3l, sorry to communicate through this, but I wonder if you received my email from Thursday?

ax3l commented:
Hi @matteodelabre, yes perfectly. I was swamped with other stuff... sorry!
